### ISSUE TITLE
error on make

### ISSUE THREAD

**Issue by gl00ten** [jesrui/hn2sqlite#1]
```
~/workspace/hackernews/hn2sqlite$ make
g++  -std=c++11 -DNDEBUG -I. -Wall -Wno-unused-local-typedefs -O2 -msse2  -lsqlite3 -s  hn2sqlite.cpp   -o hn2sqlite
/tmp/ccoXXJGT.o: In function `setupDatabase(sqlite3*) [clone .part.8]':
hn2sqlite.cpp:(.text+0x7d): undefined reference to `sqlite3_exec'
hn2sqlite.cpp:(.text+0xaa): undefined reference to `sqlite3_free'
/tmp/ccoXXJGT.o: In function `getDBUserVersion(sqlite3*, int*)':
hn2sqlite.cpp:(.text+0x10e): undefined reference to `sqlite3_exec'
hn2sqlite.cpp:(.text+0x139): undefined reference to `sqlite3_free'
/tmp/ccoXXJGT.o: In function `setDBUserVersion(sqlite3*, int)':
hn2sqlite.cpp:(.text+0x1b5): undefined reference to `sqlite3_exec'
hn2sqlite.cpp:(.text+0x1e2): undefined reference to `sqlite3_free'
/tmp/ccoXXJGT.o: In function `setupDatabase(sqlite3*)':
hn2sqlite.cpp:(.text+0x294): undefined reference to `sqlite3_errstr'
/tmp/ccoXXJGT.o: In function `insertStory(sqlite3*, Item const&)':
hn2sqlite.cpp:(.text+0x349): undefined reference to `sqlite3_prepare_v2'
hn2sqlite.cpp:(.text+0x359): undefined reference to `sqlite3_finalize'
hn2sqlite.cpp:(.text+0x391): undefined reference to `sqlite3_bind_int'
hn2sqlite.cpp:(.text+0x3b6): undefined reference to `sqlite3_bind_text'
hn2sqlite.cpp:(.text+0x3db): undefined reference to `sqlite3_bind_text'
hn2sqlite.cpp:(.text+0x404): undefined reference to `sqlite3_bind_text'
hn2sqlite.cpp:(.text+0x423): undefined reference to `sqlite3_bind_int'
hn2sqlite.cpp:(.text+0x44f): undefined reference to `sqlite3_bind_text'
hn2sqlite.cpp:(.text+0x46e): undefined reference to `sqlite3_bind_int'
hn2sqlite.cpp:(.text+0x48d): undefined reference to `sqlite3_bind_int'
hn2sqlite.cpp:(.text+0x4a1): undefined reference to `sqlite3_step'
/tmp/ccoXXJGT.o: In function `insertComment(sqlite3*, Item const&)':
hn2sqlite.cpp:(.text+0x50f): undefined reference to `sqlite3_prepare_v2'
hn2sqlite.cpp:(.text+0x51f): undefined reference to `sqlite3_finalize'
hn2sqlite.cpp:(.text+0x561): undefined reference to `sqlite3_bind_int'
hn2sqlite.cpp:(.text+0x586): undefined reference to `sqlite3_bind_text'
hn2sqlite.cpp:(.text+0x5a1): undefined reference to `sqlite3_bind_int'
hn2sqlite.cpp:(.text+0x5cd): undefined reference to `sqlite3_bind_text'
hn2sqlite.cpp:(.text+0x5ec): undefined reference to `sqlite3_bind_int'
hn2sqlite.cpp:(.text+0x60b): undefined reference to `sqlite3_bind_int'
hn2sqlite.cpp:(.text+0x62a): undefined reference to `sqlite3_bind_int'
hn2sqlite.cpp:(.text+0x63e): undefined reference to `sqlite3_step'
/tmp/ccoXXJGT.o: In function `void rapidjson::GenericReader<rapidjson::UTF8<char>, rapidjson::UTF8<char>, rapidjson::MemoryPoolAllocator<rapidjson::CrtAllocator> >::ParseObject<2u, rapidjson::FileReadStream, ItemsHandler<rapidjson::UTF8<char> > >(rapidjson::FileReadStream&, ItemsHandler<rapidjson::UTF8<char> >&)':
hn2sqlite.cpp:(.text._ZN9rapidjson13GenericReaderINS_4UTF8IcEES2_NS_19MemoryPoolAllocatorINS_12CrtAllocatorEEEE11ParseObjectILj2ENS_14FileReadStreamE12ItemsHandlerIS2_EEEvRT0_RT1_[_ZN9rapidjson13GenericReaderINS_4UTF8IcEES2_NS_19MemoryPoolAllocatorINS_12CrtAllocatorEEEE11ParseObjectILj2ENS_14FileReadStreamE12ItemsHandlerIS2_EEEvRT0_RT1_]+0x8c5): undefined reference to `sqlite3_errmsg'
/tmp/ccoXXJGT.o: In function `main':
hn2sqlite.cpp:(.text.startup+0x35): undefined reference to `sqlite3_open'
hn2sqlite.cpp:(.text.startup+0x29c): undefined reference to `sqlite3_exec'
hn2sqlite.cpp:(.text.startup+0x303): undefined reference to `sqlite3_exec'
hn2sqlite.cpp:(.text.startup+0x332): undefined reference to `sqlite3_free'
hn2sqlite.cpp:(.text.startup+0x33c): undefined reference to `sqlite3_close'
hn2sqlite.cpp:(.text.startup+0x36a): undefined reference to `sqlite3_close'
hn2sqlite.cpp:(.text.startup+0x39b): undefined reference to `sqlite3_errmsg'
hn2sqlite.cpp:(.text.startup+0x3c0): undefined reference to `sqlite3_close'
hn2sqlite.cpp:(.text.startup+0x3f3): undefined reference to `sqlite3_free'
hn2sqlite.cpp:(.text.startup+0x402): undefined reference to `sqlite3_close'
hn2sqlite.cpp:(.text.startup+0x416): undefined reference to `sqlite3_close'
collect2: error: ld returned 1 exit status
<builtin>: recipe for target 'hn2sqlite' failed
make: *** [hn2sqlite] Error 1
```

**Comment by jesrui** [comment]
Can't reproduce your error. I am using sqlite version 3.19.3 on linux
and hn2sqlite compiles fine here.

What version of sqlite are you using? What OS? Can you check if it is correctly
installed by linking a simple program against it?

**Comment by gl00ten** [comment]
$ sqlite3 -version
3.11.0 2016-02-15 17:29:24 3d862f207e3adc00f78066799ac5a8c282430a5f

ubuntu 16.04

Can you check if it is correctly installed by linking a simple program against it? <- how you do this? :)

**Comment by jesrui** [comment]
You were right. The link order given in the Makefile was wrong. The program did compile here, but Ubuntu is less permissive with that. Please try the last version.

**Comment by gl00ten** [comment]
Ty ty. I ended up parsing the dump into a python dict, I think. But thanks for fixing this :)

**Comment by jesrui** [comment]
Fixed in  283202c.
